tracer/test/test-applications/integrations/Samples.InstrumentedTests/Vulnerabilities/WeakRandomness/RandomTests.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using FluentAssertions;
+using Xunit;
+
+namespace Samples.InstrumentedTests.Iast.Vulnerabilities.SSRF;
+public class TestClass : Random
+{
+    public TestClass()
+    {
+    }
+}
+
+public class RandomTests : InstrumentationTestsBase
+{
+    public RandomTests()
+    {
+    }
+
+    [Fact]
+    public void GivenATestClassDerivedFromRandom_WhenCallingConstructor_VulnerabilityIsReported()
+    {
+        _ = new TestClass();
```

Comment:
I'm just wondering _why_ this is considered insecure 🤔 Sure, if you're using it for crypto stuff, it'll be an issue, but if you use it in a thread-safe way it should be ok right? Seems like it could give false positives? 

To be clear, I get that calling `new Random().Next()` in a controller or something is an issue, but if you only call `new Random()` once in your app, and access it in a thread-safe way, it should be ok for non-crypto right? 🤔 

Obviously I'm not a security person, just wondering really 😄

Reply:
That's a classic discussion and a valid point. It depends on the context. If you are using it for crypto, it's a issue. We don't really have a way to detect if the generated number is used or not for sensitive stuff, so we launch the vulnerability. Would it cause false positives? Sure it does. It is actually the most reported vulnerability in Java (the one that has already implemented it). This topic is a regular subject of discussion in IAST. The current approach is to raise a low severity vulnerability and let the client check it and close it if it's a false positive.